runtime/lua/vim/ui.lua

```diff
@@ -9,6 +9,14 @@ local M = {}
 ---     - format_item (function item -> text)
 ---               Function to format an
 ---               individual item from `items`. Defaults to `tostring`.
+---     - kind (string|nil)
+---               Kind of `items`.
+---               Plugins reimplementing `vim.ui.select` may wish to
+---               display different ui elements depending on the
+---               the structure of `items`, or the context in which
+---               `select` is called. `kind` allows authors
+---               of such plugins to opt into maintaining such
+---               behavior.
```

Comment:
Can say  more, with less.

```suggestion
---     - kind (string|nil)
---               Arbitrary hint string indicating the item shape.
---               Plugins reimplementing `vim.ui.select` may wish to
---               use this to infer the structure or semantics of
---               `items`, or the context in which select() was called.
```